README.md

```diff
@@ -186,8 +186,16 @@ It's also really easy to implement your own caching strategy, by extending the I
 export abstract class IStorageStrategy {
   abstract getAll(cacheKey: string): Array<ICachePair<any>>;
   abstract add(entity: ICachePair<any>, cacheKey: string): void;
+  /**
+   * @deprecated Use update instead.
+   */
   abstract updateAtIndex(index: number, entity: ICachePair<any>, cacheKey: string): void;
+  abstract update?(index: number, entity: ICachePair<any>, cacheKey: string): void;
+  /**
+   * @deprecated Use remove instead.
+   */
   abstract removeAtIndex(index: number, cacheKey: string): void;
+  abstract remove?(index: number, entity: ICachePair<any>, cacheKey: string): void;
```

Comment:
Do you think that `index` is stil needed here ? 
1. If you want you will search for index in implementation like you do `cacheList.indexOf(entity)`
2. In other scenarios ( like mine ) I don't need to search for it at all
That's only a suggestion to consider, but I also see a value to have it for convenience.

Reply:
I've kept the index because it's hard to search within non-referential storages like Local Storage for example. With the index, you can do a fast search on that.